src/backend/cuda/platform.cpp

```diff
@@ -379,6 +379,24 @@ BlasHandle blasHandle() {
     return *cublasManager(cuda::getActiveDeviceId());
 }
 
+unique_handle<cudnnHandle_t>& nnManager(const int deviceId) {
+    thread_local unique_handle<cudnnHandle_t>
+        cudnnHandles[DeviceManager::MAX_DEVICES];
+    thread_local std::once_flag initFlags[DeviceManager::MAX_DEVICES];
+
+    std::call_once(initFlags[deviceId],
+                   [&] { cudnnHandles[deviceId].create(); });
+
+    CUDNN_CHECK(
+        cudnnSetStream(cudnnHandles[deviceId], cuda::getStream(deviceId)));
+
+    return cudnnHandles[deviceId];
+}
+
+cudnnHandle_t nnHandle() {
+    return nnManager(cuda::getActiveDeviceId());
+}
+
```

Comment:
We should reset/release the cudnn resources inside `~DeviceManager` like we do for other cublas/cusolver/cusparse resource handlers .

https://github.com/arrayfire/arrayfire/blob/0d6fc70dab7d4882ef9d2d5ddf6ad2f71d18dda7/src/backend/cuda/platform.cpp#L144